packages/grub2/config/grub.cfg.tmpl

```diff
@@ -1,27 +1,15 @@
 set timeout=10
 
-# Load custom env file
-set env_file="/grubenv"
-search --no-floppy --file --set=env_blk "${env_file}"
+# Load env file if file exists and size its not 0
+search --no-floppy --label --set=customenv @STATE_LABEL@
```

Comment:
Why no call it `state_blk` instead of `customenv`? We can also reuse this variable in cos and fallback menus. So we also skip the search in there.

Reply:
I am wondering if at this stage $root is already set, probably it is essentially set to the current blk device. If so this is already state and there is no need to re-search it again (it was already searched to read the cfg file). So probably we can even remove this one and relay on $root.

Reply:
Well I had variable issues with reusing vars in the grub.cfg file when I was testing some other things.

First I want to check that this passes the tests, then I will have a look at this!

Reply:
According to the grub2 [manual](https://www.gnu.org/software/grub/manual/grub/grub.html#config_005fdirectory) we should have already properly set the `config_directory` after loading the config file. So files could be checked under this directory and event avoid the STATE search. This is relevant as it allows us to choose the STATE label at install time without having to touch the in image `grub.cfg`.

Reply:
oh wow, that is handy...so then we only need to change the root if we are on recovery?? Pretty nice!